iceoryx_utils/test/moduletests/test_posix_timer.cpp

```diff
@@ -65,40 +65,62 @@ TEST_F(TimerStopWatch_test, DurationOfZeroCausesError)
+    TIMING_TEST_EXPECT_FALSE(sut.hasExpired());
     std::this_thread::sleep_for(std::chrono::milliseconds(2 * TIMEOUT.milliSeconds<int>() / 3));
-    TIMING_TEST_EXPECT_FALSE(sut.hasExpiredComparedToCreationTime());
+    TIMING_TEST_EXPECT_FALSE(sut.hasExpired());
     std::this_thread::sleep_for(std::chrono::milliseconds(2 * TIMEOUT.milliSeconds<int>() / 3));
-    TIMING_TEST_EXPECT_TRUE(sut.hasExpiredComparedToCreationTime());
+    TIMING_TEST_EXPECT_TRUE(sut.hasExpired());
 });
 
 TIMING_TEST_F(TimerStopWatch_test, ResetWithDurationIsExpired, Repeat(5), [&] {
-    Timer sut(TIMEOUT);
+    iox::clock::DeadlineTimer sut(TIMEOUT);
     std::this_thread::sleep_for(std::chrono::milliseconds(2 * TIMEOUT.milliSeconds<int>()));
-    TIMING_TEST_EXPECT_TRUE(sut.hasExpiredComparedToCreationTime());
-    sut.resetCreationTime();
-    TIMING_TEST_EXPECT_FALSE(sut.hasExpiredComparedToCreationTime());
+    TIMING_TEST_EXPECT_TRUE(sut.hasExpired());
+    sut.reset();
+    TIMING_TEST_EXPECT_FALSE(sut.hasExpired());
 });
 
 TIMING_TEST_F(TimerStopWatch_test, ResetWhenNotExpiredIsStillNotExpired, Repeat(5), [&] {
-    Timer sut(TIMEOUT);
+    iox::clock::DeadlineTimer sut(TIMEOUT);
     std::this_thread::sleep_for(std::chrono::milliseconds(2 * TIMEOUT.milliSeconds<int>() / 3));
-    sut.resetCreationTime();
+    sut.reset();
     std::this_thread::sleep_for(std::chrono::milliseconds(2 * TIMEOUT.milliSeconds<int>() / 3));
-    TIMING_TEST_EXPECT_FALSE(sut.hasExpiredComparedToCreationTime());
+    TIMING_TEST_EXPECT_FALSE(sut.hasExpired());
 });
 
 TIMING_TEST_F(TimerStopWatch_test, ResetAfterBeingExpiredIsNotExpired, Repeat(5), [&] {
-    Timer sut(TIMEOUT);
+    iox::clock::DeadlineTimer sut(TIMEOUT);
     std::this_thread::sleep_for(std::chrono::milliseconds(2 * TIMEOUT.milliSeconds<int>()));
 
-    TIMING_TEST_ASSERT_TRUE(sut.hasExpiredComparedToCreationTime());
-    sut.resetCreationTime();
-    TIMING_TEST_EXPECT_FALSE(sut.hasExpiredComparedToCreationTime());
+    TIMING_TEST_ASSERT_TRUE(sut.hasExpired());
+    sut.reset();
+    TIMING_TEST_EXPECT_FALSE(sut.hasExpired());
 });
 
+TIMING_TEST_F(TimerStopWatch_test, ResetWithCustomizedTimeAfterBeingExpiredIsNotExpired, Repeat(5), [&] {
+    iox::clock::DeadlineTimer sut(TIMEOUT);
+    std::this_thread::sleep_for(std::chrono::milliseconds(2 * TIMEOUT.milliSeconds<int>()));
+
+    TIMING_TEST_ASSERT_TRUE(sut.hasExpired());
+
+    sut.reset(20_s);
+    std::this_thread::sleep_for(std::chrono::milliseconds(2 * TIMEOUT.milliSeconds<int>()));
+
+    TIMING_TEST_EXPECT_FALSE(sut.hasExpired());
+});
+
+
+TIMING_TEST_F(TimerStopWatch_test, RemainingTimeCheckIfNotExpired, Repeat(5), [&] {
+    iox::clock::DeadlineTimer sut(20_ms);
+    std::this_thread::sleep_for(std::chrono::milliseconds(TIMEOUT.milliSeconds<int>()));
+    int remainingTime = sut.remainingTime().milliSeconds<int>();
+
+    TIMING_TEST_EXPECT_TRUE(remainingTime >= 8 && remainingTime <= 10);
```

Comment:
there are quite some magic numbers here. Could you give them names? Or maybe also calculate them like the remaining which is just the timer timeout - sleep time +/- some jitter

Reply:
Okay. I will assign meaningful names to these integers.